pulsar/math/export.cpp

```diff
@@ -58,18 +72,20 @@ void export_eigen_x_impl(pybind11::module& m,const char* Name)
 //TensorI is the type it wraps, and Name is the python class name
 template<typename TensorT,typename TensorI>
 void  export_irrep_spin_X(pybind11::module& m,const char* Name)
-{   pybind11::class_<TensorT>(m,Name)
+{   pybind11::class_<TensorT/*,std::shared_ptr<TensorT>*/>(m,Name)
+    .def(pybind11::init<>())
+    .def(pybind11::init<const TensorT&>())
     .def("has",&TensorT::has)
     .def("get_irreps",&TensorT::get_irreps)
     .def("get_spins",&TensorT::get_spins)
-    .def("get",[](TensorT& t,Irrep ir,int spin)->std::shared_ptr<TensorI>{
+    .def("get",[](TensorT& t,Irrep ir,int spin){
                   return t.get(ir,spin);          
      })
-    .def("set",[](TensorT& t,Irrep ir,int spin,TensorI&& tensor){        
-        t.set(ir,spin,std::move(std::make_shared<TensorI>(std::move(tensor))));
+    .def("set",[](TensorT& t,Irrep ir,int spin,const TensorI& tensor){        
+        t.set(ir,spin,std::move(tensor));
     })
```

Comment:
1. I think the shared_ptr has to be enabled for this class, since there are some places we use that.

2. Not sure why get/set are lambdas. You should just be able to call the class members directly

3. For 'set', you are moving from a const reference. This might be the cause of some of your problems?